src/com/hapgood/refreshable.cljc

```diff
@@ -27,17 +43,42 @@
              IWithMeta
              (-with-meta [this m'] (reset! m m'))))
   ;; Inspired by https://clojure.atlassian.net/browse/ASYNC-102
-  #?@(:clj (clojure.lang.IDeref ; This interface is semantically inappropriate for ClojureScript, right?
+  #?@(:clj (clojure.lang.IDeref
+            ; This interface is semantically inappropriate for ClojureScript, right?
             (deref [this]
                    (let [p (promise)]
                      (async/take! this (fn [x] (deliver p x)))
                      (deref p)))
             clojure.lang.IBlockingDeref
             (deref [this timeout fallback]
                    (let [t (async/timeout timeout)
-                         p (promise)
                          [val port] (async/alts!! [t this])]
-                     (if (= this port) val fallback)))))
+                     (if (= this port) val fallback)))
+            clojure.lang.IRef
+            (setValidator [this f]
+                          (println "setting validator...")
+                          (when (async/poll! this)
+                            (validate! f @this))
+                          (set! validator f)
+                          this)
+            (getValidator [_this] validator)
+            (getWatches [_this] @watches)
+            (addWatch [this k f] (swap! watches assoc k f) this)
+            (removeWatch [this k] (swap! watches dissoc k) this))
+      :cljs (IDeref
+             ; This interface is semantically inappropriate for ClojureScript, right?
+             ;   - Mostly, if we want to use set-validator! it uses -deref in ClojureScrip
+             (-deref [this]
+                     (async/poll! this))
+             IWatchable
+             (-notify-watches [this oldval newval]
```

Comment:
This seems like an odd function to expose in the protocol, but it seems like you have implemented it per the docs.  TIL.